Assets/Scripts/Pauser.cs

```diff
@@ -1,19 +1,73 @@
 ﻿using UnityEngine;
 using System.Collections;
+using UnityEngine.SceneManagement;
 
 public class Pauser : MonoBehaviour {
-	private bool paused = false;
-	
-	// Update is called once per frame
-	void Update () {
-		if(Input.GetKeyUp(KeyCode.P))
-		{
-			paused = !paused;
-		}
-
-		if(paused)
-			Time.timeScale = 0;
-		else
-			Time.timeScale = 1;
-	}
-}
+    public static bool IsPaused = false;
+
+    public GameObject PauseMenuObject;
+
+    // Use this for initialization
+    void Start()
+    {
+        PauseMenuObject.SetActive(IsPaused);
```

Comment:
Is this necessary?

Reply:
No indeed. This same code is in the Resume() method ... Called just after this line. I will correct that.

However the Resume() method in Start() is necessary to ensure the game is no longer paused. (When we enter in a level, we go to the menu and we return to the same level)

Reply:
Another mistake on my part is to have used the variable "isPaused" here while I have to use the one from Settings Service. I correct that too.